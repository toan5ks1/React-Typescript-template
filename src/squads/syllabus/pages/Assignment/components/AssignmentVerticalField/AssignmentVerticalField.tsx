import { Box } from "@mui/material";
import TypographyBase, { TypographyBaseProps } from "src/components/Typographys/TypographyBase";

// Replicate current UI due to new components
const sx = {
    label: {
        textTransform: "uppercase",
        fontSize: "1rem",
    },
    value: {
        fontWeight: 500,
        fontSize: "1rem",
    },
};

interface AssignmentVerticalFieldProps {
    label: string;
    value: string | number;
    valueTypoProps?: object | TypographyBaseProps;
}

const AssignmentVerticalField = (props: AssignmentVerticalFieldProps) => {
    const { label, value, valueTypoProps } = props;

    return (
        <Box display="flex" flexDirection="column" mt={1} mb={3}>
            <Box mb={1}>
                <TypographyBase variant="h6" color="textSecondary" sx={sx.label}>
                    {label}
                </TypographyBase>
            </Box>
            <TypographyBase variant="body2" color="textPrimary" sx={sx.value} {...valueTypoProps}>
                {value}
            </TypographyBase>
        </Box>
    );
};

export default AssignmentVerticalField;
