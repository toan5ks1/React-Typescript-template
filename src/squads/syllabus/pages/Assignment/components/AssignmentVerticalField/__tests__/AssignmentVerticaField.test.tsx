import { ComponentProps } from "react";

import AssignmentVerticalField from "../AssignmentVerticalField";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<AssignmentVerticalField />", () => {
    let wrapper: RenderResult;

    const props: ComponentProps<typeof AssignmentVerticalField> = {
        label: "Name",
        value: "Test",
        valueTypoProps: {
            "data-testid": "Test_Id",
        },
    };

    beforeEach(() => {
        wrapper = render(<AssignmentVerticalField {...props} />);
    });

    it("should exist label", () => {
        expect(screen.getByText("Name")).toBeInTheDocument();
    });

    it("should exist value", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should exist data-testid", () => {
        expect(screen.getByTestId("Test_Id")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
