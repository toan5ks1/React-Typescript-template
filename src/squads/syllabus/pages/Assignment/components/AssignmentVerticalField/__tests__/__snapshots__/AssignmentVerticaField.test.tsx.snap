// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssignmentVerticalField /> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 24px;
}

.emotion-1 {
  margin-bottom: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  color: rgba(0, 0, 0, 0.6);
  text-transform: uppercase;
  font-size: 1rem;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.87);
  font-weight: 500;
  font-size: 1rem;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 emotion-2"
      >
        Name
      </h6>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-3"
      data-testid="Test_Id"
    >
      Test
    </p>
  </div>
</div>
`;
